abcd/backends/utils.py

```diff
@@ -0,0 +1,131 @@
+from collections import Counter
+from datetime import datetime
+import logging
+from operator import itemgetter
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+def histogram(name, data, **kwargs):
+    if not data:
+        return None
+
+    if data and isinstance(data, list):
```

Comment:
`not data` has already been checked. Also, what if it's a `tuple` or `numpy.ndarray`?